File: C++/Solution274.cpp
#include<iostream>
#include<vector>
#include<algorithm>


using namespace std;

class Solution274 {
public:
    int hIndex(vector<int>& citations) {
        int res = 0;
        sort(citations.begin(), citations.end());
        for(int i = citations.size() - 1; i >= 0; i--){
            int cur = citations[i];
            res = max(res, min(cur, (int) citations.size() - i));
        }
        return res;
    }
};

File: Java/other/Solution274.java
import java.util.Arrays;

class Solution274 {
    public int hIndex(int[] citations) {
    	if(citations.length == 0){
            return 0;
        }
    	int length = citations.length;
    	Arrays.sort(citations);
    	int max = Math.min(length, citations[0]);
    	for(int i = 0; i < length; i++){
    		int cur = Math.min(length - i, citations[i]);
    		max = Math.max(max, cur);
    	}
    	return max;
    }
}